partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/PlanTransform.kt

```diff
@@ -157,11 +158,9 @@ internal class PlanTransform(
         override fun visitRexOpPath(node: Rex.Op.Path, ctx: Unit) =
             super.visitRexOpPath(node, ctx) as org.partiql.plan.Rex.Op.Path
 
-        override fun visitRexOpCast(node: Rex.Op.Cast, ctx: Unit) =
-            super.visitRexOpCast(node, ctx) as org.partiql.plan.Rex.Op.Cast
-
-        override fun visitRexOpCastUnresolved(node: Rex.Op.Cast.Unresolved, ctx: Unit): PlanNode {
-            error("Unresolved cast $node")
+        @OptIn(PartiQLValueExperimental::class)
+        override fun visitRexOpCastUnresolved(node: Rex.Op.Cast.Unresolved, ctx: Unit): org.partiql.plan.Rex.Op {
+            return rexOpErr("Unresolved cast: ${node.arg.type} to ${node.target}.", emptyList())
```

Comment:
I don't think the public plan should have error nodes btw... I suggest PlanTransform use the antlr style error listeners. It's a bit silly to hide errors in a tree, then make the compiler later throw them.

We have _two_ error nodes at the moment, but should only have _one_.

We have,
 - rexOpError
 - rexOpMissing

The _rexOpError_ should NOT exist in the public plan IR, but _rexOpMissing_ should. The _rexOpMissing_ operation represents a typing error that either throws or becomes MISSING in the evaluator.

The _rexOpError_ is an outright error that will _never_ compile.